### ISSUE TITLE
System.Memory doesn't work on UWP

### ISSUE THREAD

**Issue by eerhardt** [dotnet/corefx#25024]
Create a new blank UWP app.
Add System.Memory Version="4.5.0-preview1-25831-01" nuget package.

Write some code
```
Span<int> foo = new int[5];
```
Build.

This doesn't compile, saying there is no "Span" type.

Now add a reference to the latest UWP package:

```
    <PackageReference Include="Microsoft.NETCore.UniversalWindowsPlatform">
      <Version>6.1.0-preview1-25902-01</Version>
    </PackageReference>
```

which has the ref type for Span.

Build.

You get an error:

```
Payload contains two or more files with the same destination path 'System.Memory.dll'. Source files: C:\Users\eerhardt\.nuget\packages\runtime.win10-x64.microsoft.netcore.universalwindowsplatform\6.1.0-preview1-25902-01\runtimes\win10-x64\lib\uap10.0.15138\System.Memory.dllC:\Users\eerhardt\.nuget\packages\system.memory\4.5.0-preview1-25831-01\runtimes\win\lib\uap10.0.15138\System.Memory.dll
```

/cc @ahsonkhan @KrzysztofCwalina

**Comment by eerhardt** [comment]
Note that when using the Fall Creators update, the System.Memory assets for the `uap10.0.15138` TFM will get used.  The ref assembly in `.nuget\packages\system.memory\4.5.0-preview1-25831-01\ref\uap10.0.15138\System.Memory.dll` has a TypeForwardTo for Span and ReadOnlySpan to System.Runtime.dll.  But for `UniversalWindowsPlatform` version `6.0.1`, Span and ReadOnlySpan don't exist in that version of System.Runtime.dll.

**Comment by joshfree** [comment]
@joperezr could you help take a quick look at the package authoring?

**Comment by joperezr** [comment]
@eerhardt which version of VS are you using to compile? From what I can tell, it looks like you don't have the right conflict resolution logic that would fix the nuget clash.

**Comment by eerhardt** [comment]
@joperezr - there are 2 issues here.  The first one has nothing to do with conflict resolution.

### Issue 1:

Adding `System.Memory` as a NuGet package to a new UWP app doesn't let me reference the `System.Span<>` type.  `System.Memory` says it supports `netstandard1.1`, which UWP 10.0 implements.  I stated the reason why this doesn't work above:

> Note that when using the Fall Creators update, the System.Memory assets for the uap10.0.15138 TFM will get used. The ref assembly in .nuget\packages\system.memory\4.5.0-preview1-25831-01\ref\uap10.0.15138\System.Memory.dll has a TypeForwardTo for Span and ReadOnlySpan to System.Runtime.dll. But for UniversalWindowsPlatform version 6.0.1, Span and ReadOnlySpan don't exist in that version of System.Runtime.dll.

However, this should work because we are saying the package is `netstandard1.1`.

### Issue 2:

Updating to the latest nightly build of `Microsoft.NETCore.UniversalWindowsPlatform` produces the `Payload contains two or more files with the same destination path` error.

I have reproduced both of these issues with:

1. Microsoft Visual Studio Enterprise 2017 Int Preview
Version 15.6.0 Preview 1.1 [27228.1.d15.6]  (the latest in the `Int Preview` channel as of this morning)
2. System.Memory `4.5.0-preview1-26102-01`
3. Microsoft.NETCore.UniversalWindowsPlatform `6.1.0-preview1-26102-01`.

**Comment by eerhardt** [comment]
> they could potentially reference a different package that has a component which is .NETStandard-based, and that package could have a reference to System.Memory which will cause issue 2 to come back again.

This is exactly how I ran into this problem.  I was trying to use [System.Numerics.Tensors](https://dotnet.myget.org/feed/dotnet-corefxlab/package/nuget/System.Numerics.Tensors) (which is netstandard1.1 and has a dependency on System.Memory) in a UWP app and it doesn't work.

**Comment by joperezr** [comment]
Yeah, I kept looking at it and I finally know why Issue 1 is happening, and that is because you are referencing a package built out in master branch, which has a UAP asset for vNext but we haven't yet updated the TFM for UAP vNext, so the package is still being created with the asset being for UAP RS3, which is now the one selected and why you are seeing the issue. I created an issue that will track the update to the new TFM that should fix Issue 1.

**Comment by joperezr** [comment]
@joshfree we need to merge #28340. After that this should be closed.
